tinkerbell/stack/templates/nginx.yaml

```diff
@@ -61,23 +64,24 @@ spec:
       initContainers:
       - name: init-hook-download
         image: {{ .Values.stack.hook.image }}
-        command: ["/bin/sh", "-xec"]
-        args: # TODO(jacobweinstock): add checksum verification after download
-        - rm -rf /usr/share/nginx/html/checksums.txt;
-          touch /usr/share/nginx/html/checksums.txt;
+        command: ["/bin/bash", "-xeuc"]
+        args:
+        - |
+          apt-get update
+          apt-get install -y wget
+          cd /usr/share/nginx/html
           {{- range $index, $keys := .Values.stack.hook.downloads }}
-          echo "{{ $keys.sha512sum.kernel }}" >> /usr/share/nginx/html/checksums.txt;
-          echo "{{ $keys.sha512sum.initramfs }}" >> /usr/share/nginx/html/checksums.txt;
+          wget -O /tmp/hook{{ $index }}.tar.gz {{ $keys.url }}
+          tar -zxvf /tmp/hook{{ $index }}.tar.gz
+          rm /tmp/hook{{ $index }}.tar.gz
```

Comment:
I appreciate the old code does this, but if we remove the tarball the `sha512sum --check` call is useless as it won't have anything to compute against the checksum in `checksum.txt`

Reply:
That's the strange part about the logic. It's checking the checksum of the files inside the tarball, not the tarball itself.

I would only check the tarball itself, but that change would be a breaking change. Should I do it?

Reply:
@chrisdoherty4 this would break:

https://github.com/tinkerbell/charts/blob/85832245f08a6f6a2a4cba2379267a1a1b613bd5/tinkerbell/stack/values.yaml#L17C1-L25

As it needs to be changed into:

```yaml
    downloads:
      - url: https://github.com/tinkerbell/hook/releases/download/v0.8.0/hook_x86_64.tar.gz
        sha512sum: <hook_x86_64.tar.gz sha512 checksum would go here>
      - url: https://github.com/tinkerbell/hook/releases/download/v0.8.0/hook_aarch64.tar.gz
        sha512sum: <hook_aarch64.tar.gz sha512 checksum would go here>
```

Reply:
I'm good with this breaking change. Just checksum the tarballs. 👍🏼

Reply:
Thanks for clarifying @rgl. We aren't worried, for now, about that kind of breakage.

Reply:
Please check it now.